packages/swingset-liveslots/src/virtualObjectManager.js

```diff
@@ -235,21 +246,35 @@ function checkAndUpdateFacetiousness(tag, desc, facetNames) {
 // We only need to do this for multi-facet Kinds; single-facet kinds don't
 // have any extra objects for userspace to work with.
 
-function makeRepresentative(proto) {
+const makeRepresentative = (proto, baseRef) => {
   const self = { __proto__: proto };
+  if (LABEL_INSTANCES) {
+    defineProperty(self, Symbol.toStringTag, {
+      value: `${proto[Symbol.toStringTag]}#${baseRef}`,
```

Comment:
If the creator of the vat decides to enable LABEL_INSTANCES, then code inside the vat gets to snoop on other code inside the vat (learning vrefs, and therefore how many instances of virtual/durable Kinds have been created), which is something we generally avoid. This is a debugging feature, so it's fine to make this kind of tradeoff, but when you find a place to document this, let's be sure to point out the confidentiality-loss cost.

Might be worth a one-line comment here, to remind us later.

> note: exposing vrefs to userspace reveals confidential object-creation activity

Reply:
Good point!

Done, with further clarification.